notebooks/huggingface_notebook.ipynb

```diff
@@ -0,0 +1,296 @@
+    "train_dataloader = DataLoader(train_dataset, batch_size=16, shuffle=False, drop_last=False, collate_fn=data_collator)\n",
+    "eval_dataloader = DataLoader(eval_dataset,batch_size=16, shuffle=False, drop_last=False, collate_fn=data_collator)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# define metrics and convert HF pretrained model into ComposerModel"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "metrics = [LanguageCrossEntropy(vocab_size=tokenizer.vocab_size), Accuracy()]\n",
+    "# Package as a composer model\n",
+    "composer_model = HuggingFaceModel(model, metrics=metrics)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Optimizers and Learning Rate Schedulers"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The last setup step is to create an optimizer and a learning rate scheduler. We will use PyTorch's AdamW optimizer and linear learning rate scheduler since these are typically used to fine-tune BERT on tasks such as SST-2."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from torch.optim import AdamW\n",
+    "from torch.optim.lr_scheduler import LinearLR\n",
+    "\n",
+    "optimizer = AdamW(\n",
+    "    params=composer_model.parameters(),\n",
+    "    lr=3e-5, betas=(0.9, 0.98),\n",
+    "    eps=1e-6, weight_decay=3e-6\n",
+    ")\n",
+    "linear_lr_decay = LinearLR(\n",
+    "    optimizer, start_factor=1.0,\n",
+    "    end_factor=0, total_iters=150\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Composer Trainer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will now specify a Composer `Trainer` object and run our training! `Trainer` has many arguments that are described in our [documentation](https://docs.mosaicml.com/en/stable/api_reference/composer.trainer.trainer.html), but let's discuss the less obvious arguments used below:\n",
+    "- `max_duration` - a string specifying how long to train, either in terms of batches (e.g. '10ba' is 10 batches) or epochs (e.g. '1ep' is 1 epoch).\n",
+    "- `schedulers` - a list of PyTorch learning rate schedulers that will be composed together.\n",
+    "- `device` - specifies if the training will be done on CPU or GPU by using 'cpu' or 'gpu', respectively.\n",
+    "- `train_subset_num_batches` - specifies the number of training batches to use for each epoch. This is not a necessary argument but is useful for quickly testing code.\n",
+    "- `precision` - whether to do the training in full precision 'fp32' or mixed precision 'amp'. Mixed precision provides an almost 2x speedup in training time on certain hardware. If you get a P100, try `precision='amp'`!\n",
```

Comment:
P100 doesn't have tensor cores; can just rm this sentence

Reply:
This should also replace the current `nlp` notebook correct?

Reply:
Yeah it is. I can change it in both.